src/Iact1dUnbinnedLkl.cc

```diff
@@ -889,6 +889,78 @@ Int_t Iact1dUnbinnedLkl::ReaddNdESignal(TString filename)
   return status;
 }
 
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+// 
+// Read multiple dN_i/dE for signal from file in
+// the Segue Stereo input format produced by Jelena
+// and their corresponding branching ratios BR_i.
+// Then perform a linear combination
+// dN/dE = sum(BR_i * dN_i/dE).
+// Replacement of existing file is allowed
+// Return 0 in case of success
+//        1 if file is not found
+//
+Int_t Iact1dUnbinnedLkl::ReaddNdESignal(Int_t nFiles, TString* filenames, Float_t* branchingRatios)
+{
+  // Initialize the array of pointers and other variables
+  TFile* dNdESignalFile[nFiles];
```

Comment:
I don't think you need an array of pointer for dNdESignalFile or hProvdNdESignal, since they are used only inside the loop over nFiles, therefore you can have just one pointer and reuse it at each step of the loop.

Reply:
Thank you for the additional comments! Yes, dNdESignalFile don't need to be an array of pointer. But hProvdNdESignal have to be an array of pointer. It's used in the nFiles for loop at line 949.

Reply:
The second loop could be part of the first loop

Reply:
Good point!